src/sparseml/pytorch/yolov8/trainers.py

```diff
@@ -138,32 +130,80 @@ def train(self):
-                cfg = ckpt["model"].yaml
-        else:
-            cfg = model
-        self.model = self.get_model(cfg=cfg, weights=weights)
+        if not str(self.model).endswith(".pt"):
+            # not a checkpoint - use ultralytics loading logic
+            self.model = self.get_model(cfg=self.model, weights=None)
+            self._build_managers(ckpt=None)
+            return
+
+        if not os.path.exists(str(self.model)):
+            # remote ultralytics checkpoint - zoo checkpoint was already downloaded
+            # in constructor
+            LOGGER.info("Loading remote ultralytics checkpoint")
+            weights, ckpt = attempt_load_one_weight(self.model)
+            cfg = ckpt["model"].yaml
+            self.model = self.get_model(cfg=cfg, weights=weights)
+            self._build_managers(ckpt=ckpt)
+            return ckpt
+
+        ckpt = torch.load(str(self.model), map_location="cpu")
+
+        if not ("source" in ckpt and ckpt["source"] == "sparseml"):
+            # local ultralyltics checkpoint
+            LOGGER.info("Loading local ultralytics checkpoint")
+            weights, ckpt = attempt_load_one_weight(self.model)
+            cfg = ckpt["model"].yaml
+            self.model = self.get_model(cfg=cfg, weights=weights)
+            self._build_managers(ckpt=ckpt)
+            return ckpt
+
+        # sanity check - this is one of our checkpoints
+        LOGGER.info("Loading local sparseml checkpoint")
+        assert ckpt["source"] == "sparseml"
+
+        self.model = self.get_model(cfg=ckpt["model_yaml"], weights=None)
+
+        # NOTE: this will apply structure, we need to do this before loading state dict
+        self._build_managers(ckpt=ckpt)
+
+        self.model.load_state_dict(ckpt["model"])
+        LOGGER.info("Loaded previous weights from sparseml checkpoint")
         return ckpt
 
+    def _build_managers(self, ckpt: Optional[dict]):
+        if self.args.recipe is not None:
+            self.manager = ScheduledModifierManager.from_yaml(
+                self.args.recipe, recipe_variables=self.args.recipe_args
+            )
+
+        if ckpt is None:
+            return
+
+        if "recipe" not in ckpt:
+            return
+
+        if ckpt["epoch"] == -1:
+            LOGGER.info(
+                "Applying structure from completed recipe in checkpoint "
+                f"at epoch {ckpt['epoch']}"
+            )
+            self.checkpoint_manager = ScheduledModifierManager.from_yaml(ckpt["recipe"])
+            self.checkpoint_manager.apply_structure(self.model, epoch=float("inf"))
+
+        else:
+            # resuming
+            # yolo will populate this when the --resume flag
+            assert self.args.recipe is not None
+            LOGGER.info(
+                "Applying structure from un-finished recipe in checkpoint "
+                f"at epoch {ckpt['epoch']}"
+            )
+            self.manager.apply_structure(self.model, epoch=ckpt["epoch"])
```

Comment:
instead of apply structure, if we are resuming why not initialize the manager to the epoch we are resuming at?